pip/req/req_set.py

```diff
@@ -238,6 +238,15 @@ def add_requirement(self, install_req, parent_req_name=None):
                 existing_req = None
             if (parent_req_name is None and existing_req and not
                     existing_req.constraint):
+                if existing_req.name == install_req.name:
```

Comment:
Since `existing_req = self.get_requirement(name)` both should have the same name, so I don't understand this condition...